partiql-ast/src/pretty.rs

```diff
@@ -997,6 +997,238 @@ impl PrettyDoc for Join {
+        if let Some(selector) = selector {
+            arena.intersperse([selector, patterns], arena.softline())
+        } else {
+            patterns
+        }
+        .group()
+    }
+}
+
+impl PrettyDoc for GraphMatchPattern {
+    fn pretty_doc<'b, D, A>(&'b self, arena: &'b D) -> DocBuilder<'b, D, A>
+    where
+        D: DocAllocator<'b, A>,
+        D::Doc: Clone,
+        A: Clone,
+    {
+        let pattern = pretty_seperated(
+            arena.nil().append(arena.space()),
+            &self.parts,
+            PRETTY_INDENT_MINOR_NEST,
+            arena,
+        );
+        let mut doc = if let Some(r) = &self.restrictor {
+            match r {
+                GraphMatchRestrictor::Trail => arena.text("TRAIL"),
+                GraphMatchRestrictor::Acyclic => arena.text("ACYCLIC"),
+                GraphMatchRestrictor::Simple => arena.text("SIMPLE"),
+            }
+            .append(arena.space())
+        } else {
+            arena.nil()
+        };
+
+        let pattern = if let Some(v) = &self.variable {
+            arena.intersperse(
+                [arena.text(&v.value), arena.text("="), pattern],
+                arena.space(),
+            )
+        } else {
+            pattern
+        };
+        doc = doc.append(pattern);
+
+        let brackets =
+            self.restrictor.is_some() || self.quantifier.is_some() || self.prefilter.is_some();
+
+        if brackets {
+            let doc = if let Some(filter) = &self.prefilter {
+                arena.intersperse(
+                    [doc, arena.text("WHERE"), filter.pretty_doc(arena)],
+                    arena.space(),
+                )
+            } else {
+                doc
+            };
+            let bracketed = pretty_bracketed_doc(doc, arena);
+            if let Some(postfix) = &self.quantifier {
+                arena.concat([bracketed, postfix.pretty_doc(arena)])
+            } else {
+                bracketed
+            }
+        } else {
+            doc
+        }
+        .group()
+    }
+}
+
+impl PrettyDoc for GraphMatchPatternPart {
+    fn pretty_doc<'b, D, A>(&'b self, arena: &'b D) -> DocBuilder<'b, D, A>
+    where
+        D: DocAllocator<'b, A>,
+        D::Doc: Clone,
+        A: Clone,
+    {
+        match self {
+            GraphMatchPatternPart::Node(n) => n.pretty_doc(arena),
+            GraphMatchPatternPart::Edge(e) => e.pretty_doc(arena),
+            GraphMatchPatternPart::Pattern(p) => p.pretty_doc(arena),
+        }
+    }
+}
+
+impl PrettyDoc for GraphMatchNode {
+    fn pretty_doc<'b, D, A>(&'b self, arena: &'b D) -> DocBuilder<'b, D, A>
+    where
+        D: DocAllocator<'b, A>,
+        D::Doc: Clone,
+        A: Clone,
+    {
+        let mut spec = arena.nil();
+        if let Some(r) = &self.variable {
+            spec = spec.append(arena.text(&r.value));
+        }
+        if let Some(l) = &self.label {
+            debug_assert_eq!(l.len(), 1); // TODO
```

Comment:
is this todo + `debug_assert_eq` still needed?

Reply:
Currently, parsing doesn't support more than one pattern, so yes, it's still TODO